core-web/libs/template-builder/src/lib/components/template-builder/template-builder.component.ts

```diff
@@ -406,22 +407,35 @@ export class TemplateBuilderComponent implements OnInit, AfterViewInit, OnDestro
     }
 
     /**
-     * @description This method sets the box initial values everytime a mouse leaves a row
+     * @description This method sets the box initial values everytime a mouse leaves or moves in a row
      * so that way we always have the correct value setted and overriding the behavior of gridstack
      *
      * @private
      * @param {Element} el
      * @memberof TemplateBuilderComponent
      */
-    private fixGridstackNodeOnMouseLeave(el: Element): void {
-        // So every time the mouse leaves the row, we set the initial values for the box
-        el.addEventListener('mouseleave', () => {
-            if (this.addBoxIsDragging && this.addBox.nativeElement.gridstackNode?.w !== BOX_WIDTH) {
-                this.addBox.nativeElement.gridstackNode = {
-                    ...this.addBox.nativeElement.gridstackNode,
-                    ...this.boxOptions
-                };
-            }
+    private listenToRowMouseEvents(el: Element): void {
+        // So every time the mouse leaves or moves in the row, we set the initial values for the box
+
+        ['mouseleave', 'mousemove'].forEach((eventName) => {
```

Comment:
Did you try to listen this events to the parent of the whole thing `.template-builder__main` and check the event.target to call the `fixGridStackNodeOptions`? And that way we just listen once.

Reply:
Used a similar approach